### ISSUE TITLE
Qubino (Goap) ZMNHTDx Smart meter S4 S5 S6+ switches not visible

### ISSUE THREAD

**Issue by redswan01** [domoticz/domoticz#3247]
This leads me to another query which maybe rather for @Fishwaldo, this smart meter also has two switches which don't appear in the UI even though they are enabled in the config.

![Node10 switches](https://user-images.githubusercontent.com/7438646/57945335-e9f69b00-78d9-11e9-8aae-44b3773a21f1.PNG)

_Originally posted by @redswan01 in https://github.com/domoticz/domoticz/issues/3213#issuecomment-493535952_

**Comment by redswan01** [comment]
Just back home hence delay in replying.

Thanks to @Sneeuwtijger for indicating how to enable debug.

@gizmocuz , @Fishwaldo, here's the OZW_Log which I hope now contains some useful information.

[OZW_Log.zip](https://github.com/domoticz/domoticz/files/3198513/OZW_Log.zip)

with 10798 installed

**Comment by redswan01** [comment]
@gizmocuz , @Fishwaldo, have you had time to look into this? Looking at the log file it would appear that the device doesn't reply with any of the switch command classes, I would have expected to see COMMAND_CLASS_SWITCH_BINARY or COMMAND_CLASS_SWITCH_ALL. I'm intrigued as, according to the Qubino [compatibility list](https://qubino.com/products/smart-meter/smart-meter-compatibility/), at least SWITCH_ALL should be available. Also why are the switches not available in Domoticz when they are apparently in other systems? Is this a limitation of OZW?

**Comment by gizmocuz** [comment]
@redswan01 , Fishwaldo is not a domoticz support engineer. First check if they are available in the control panel.
Second, we do not implement the switch_all

It might be best if you update to the stable version if you can not debug/compile/breakpoint/test yourself

**Comment by redswan01** [comment]
@gizmocuz , 
> Fishwaldo is not a domoticz support engineer.

OK I'll open a new issue under OZW

> First check if they are available in the control panel.

No they're not available in the control panel

> Second, we do not implement the switch_all

Not a problem for me, I'd rather be able to control them individually .

> It might be best if you update to the stable version if you can not debug/compile/breakpoint/test yourself

I'm in the beta path because info given by the author of a python plugin I use, for RFPlayer, says that it is necessary as the plugin framework is not available in the stable release. This info dates from may 2017 so is maybe no longer the case I will check this out and move to stable if possible.

Thanks for your help and this great system. 👍 👍

I'll close this issue

**Comment by Fishwaldo** [comment]
Did you, as the documentation says - Exclude and Include the device after changing that parameter? 

The Log file doesn't help much, as you need to remove ozwcache_*.xml so I can see the original interview process, but as far as I can tell, the node does not advertise that it has multiple endpoints, hence we can't see those switches.

**Comment by redswan01** [comment]
@Fishwaldo , Here's what I just did:
Stopped Domoticz service.
Deleted ozwcache_*.xml
restarted Domoticz service
in Hardware disabled/enabled ZWave

Here are the resulting log and cache files: 
[OZW log and cache.zip](https://github.com/domoticz/domoticz/files/3213976/OZW.log.and.cache.zip)

Hope this sheds a bit more light on the problem.
I understand that the node doesn't appear to advertise all its capabilities, what puzzles me is if this is the case how do other systems manage to control these switches (if one believes the compatibility table)?

Should we move this issue to OZW rather than here under Domoticz?

**Comment by Fishwaldo** [comment]
ozwcache was still present in that log file you sent:
/home/pi/domoticz/Config/ozwcache_0xd0a6ebd4.xml

> I understand that the node doesn't appear to advertise all its capabilities, what puzzles me is if this is the case how do other systems manage to control these switches (if one believes the compatibility table)?

I have no idea? Ugly hack in the code for this device?

**Comment by redswan01** [comment]
@gizmocuz , @Fishwaldo , OK, I'm dumb  :blush: I completely misunderstood Fishwaldo's comment when he mentioned "exclude/include", as we were talking about logs I took this to mean "disable/enable" zw in hardware, I've just realized this was about the zw network. I completely missed this in the device's documentation the first time round. Having now done this the switches now appear and function, apart from the several seconds delay in the device reporting the changed status, as mentioned in issue [Qubino Smart Meter - Wrong switch states](https://github.com/OpenZWave/open-zwave/issues/1261#issue-238815081).
I owe you guys a thousand apologies for wasting your time when you have better things to do. As the saying goes "the bug is between the chair and the keyboard".
